services/order.py

```diff
@@ -0,0 +1,31 @@
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+from django.db import transaction
+from db.models import Ticket, Order
+
+
+@transaction.atomic
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: str = None
+                 ) -> QuerySet:

```

Comment:
The function `create_order` is quite large and does several things. This might make it harder to understand and maintain. Consider splitting it into smaller functions each doing one thing. This is a suggestion for improving code quality.